src/components/StatsPanel/StatsPanel.test.tsx

```diff
@@ -0,0 +1,15 @@
+import { render, screen } from '@testing-library/react';
+import { describe, it, expect } from 'vitest';
+
+import { StatsPanel } from './StatsPanel';
+
+describe('StatsPanel', () => {
+  it('renders the provided stats correctly', () => {
+    render(<StatsPanel totalTests={32} passed={12} failed={19} waiting={1} />);
+
+    expect(screen.getByText('32')).toBeInTheDocument();
```

Comment:
Using getByText('32') and similar numeric string matches may match other elements unexpectedly (for example if labels or other content contain the same digits). Tests should also assert labels, not only numeric values.

**Sugerencia:**
```typescript
Assert both label and value pairs, or use test ids/aria-labels to target the specific value element. Example:
expect(screen.getByText('Total tests')).toBeInTheDocument();
expect(screen.getByText('32')).toBeInTheDocument();

Or use a specific query like getByLabelText as shown in the previous suggestion.
```